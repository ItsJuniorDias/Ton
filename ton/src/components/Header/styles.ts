import styled from 'styled-components/native';
import { scale } from '../../utils';
import { getStatusBarHeight } from 'react-native-iphone-x-helper';

export const Container = styled.View`
  width: 100%;
  height: ${scale(64)};
  padding: 0px ${scale(24)};
  background-color: ${({ theme }) => theme.colors.shape};
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  margin-top: ${getStatusBarHeight()}px;
`;

export const Title = styled.Text`
  font-family: ${({ theme }) => theme.fonts.medium};
  font-size: ${scale(18)};
  color: ${({ theme }) => theme.colors.title};
`;

export const FakeView = styled.View`
  width: ${scale(26)};
  height: ${scale(26)};
`;

export const Notification = styled.View`
  width: ${scale(18)};
  height: ${scale(18)};
  background-color: ${({ theme }) => theme.colors.notification};
  border-radius: ${scale(10)};
  align-items: center;
  justify-content: center;
  position: absolute;
  margin-left: ${scale(16)};
  margin-top: ${scale(-8)};
`;

export const NotificationValue = styled.Text`
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: ${scale(14)};
  color: ${({ theme }) => theme.colors.title};
`;
