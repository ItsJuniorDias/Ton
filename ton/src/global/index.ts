export * from './shadowStyles';
export * from './theme';
