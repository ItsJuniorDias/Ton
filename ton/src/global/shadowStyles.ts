export const shadowStyles = {
  shadowColor: '#000',
  shadowOffset: {
    width: 0,
    height: 5,
  },
  shadowOpacity: 0.1,
  shadowRadius: 2.62,
  elevation: 2,
};
