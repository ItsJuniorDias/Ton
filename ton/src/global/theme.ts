export const theme = {
  colors: {
    primary: '#01ad0b',
    secondary: '#01ea34',
    shape: '#ffffff',
    background: '#eff4f8',
    title: '#000000',
    text: '#20252b',
    notification: '#ff7b00',
    alert: '#ed5e69',
  },
  fonts: {
    regular: 'Roboto-Regular',
    medium: 'Roboto-Medium',
    bold: 'Roboto-Bold',
  },
};
