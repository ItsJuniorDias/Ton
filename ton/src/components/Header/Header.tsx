import React, { useRef } from 'react';
import { Pressable } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { useNavigation } from '@react-navigation/native';
import type { NativeStackNavigationProp } from '@react-navigation/native-stack';

import { theme } from '../../global';

import {
  Container,
  Title,
  FakeView,
  Notification,
  NotificationValue,
} from './styles';
import { RootStackParamList } from '../../routes';

export type HeaderProps = {
  title: string;
  amountCart: number;
  isGoBack?: boolean;
};

export const Header = ({ title, amountCart, isGoBack }: HeaderProps) => {
  const testIDs = useRef({
    amount: 'amountCart_testID',
    pressable: 'pressableHeader_testID',
    cart: 'pressableCart_testID',
  }).current;

  const navigation =
    useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  return (
    <Container>
      {isGoBack ? (
        <Pressable
          testID={testIDs.pressable}
          onPress={() => navigation.goBack()}
          style={({ pressed }) => [
            {
              opacity: pressed ? 0.6 : 1,
            },
          ]}
        >
          <Icon name="arrow-back-ios" size={22} color={theme.colors.text} />
        </Pressable>
      ) : (
        <FakeView />
      )}

      <Title>{title}</Title>

      <Pressable
        testID={testIDs.cart}
        onPress={() => navigation.navigate('Cart')}
        style={({ pressed }) => [
          {
            opacity: pressed ? 0.6 : 1,
          },
        ]}
      >
        <Icon name="shopping-cart" size={28} color={theme.colors.primary} />

        <Notification>
          <NotificationValue testID={testIDs.amount}>
            {amountCart.toString()}
          </NotificationValue>
        </Notification>
      </Pressable>
    </Container>
  );
};
