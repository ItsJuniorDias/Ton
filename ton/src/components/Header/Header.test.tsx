import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { ThemeProvider } from 'styled-components/native';
import { Header } from './Header';
import { theme } from '../../global';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');
jest.mock('react-native-iphone-x-helper', () => ({
  getStatusBarHeight: jest.fn(),
  isIphoneX: jest.fn(),
}));

describe('Behavior Header', () => {
  const screenRender = () => (
    <ThemeProvider theme={theme}>
      <NavigationContainer>
        <Header amountCart={2} title="Header" />
      </NavigationContainer>
    </ThemeProvider>
  );

  it('render snapshot', () => {
    const result = render(screenRender()).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should render Header with amountCard', () => {
    const { getByTestId } = render(screenRender());

    const amount = getByTestId('amountCart_testID');

    expect(amount.props.children).toEqual('2');
  });
});
