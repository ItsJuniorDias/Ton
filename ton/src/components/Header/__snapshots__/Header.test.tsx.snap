// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior Header render snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "height": 188,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 47,
        "paddingRight": 47,
        "paddingTop": "px",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "height": 51,
          "width": 51,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Roboto-Medium",
          "fontSize": 35,
        },
      ]
    }
  >
    Header
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
      ]
    }
  >
    <Icon
      color="#01ad0b"
      name="shopping-cart"
      size={28}
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#ff7b00",
            "borderRadius": 20,
            "height": 35,
            "justifyContent": "center",
            "marginLeft": 31,
            "marginTop": -16,
            "position": "absolute",
            "width": 35,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Roboto-Regular",
              "fontSize": 27,
            },
          ]
        }
        testID="amountCart_testID"
      >
        2
      </Text>
    </View>
  </View>
</View>
`;
